File: c++/DesignPattern/Singleton/Bread.cpp
#include <iostream>
#include <string>
using namespace std;

class Bread{
    
    public:
    void  ChargeEnergy(){
     
     cout<< "Charing " << "Energy";
     
    }
   
     
};
int main(){

  Bread bread;
  cout<< Bread::s<< endl;
  bread.ChargeEnergy();
 
}

File: c#/DesignPattern/Singleton/Bread.cs
 
 
public class Bread{
 
   private static Bread bread=new Bread();
      
     public Bread(){
         System.Console.WriteLine("delicious bread  ready  ");
     } 
     public static  Bread GetInstance(){

       return  bread;
   }   
    public void ChargeEnergy(){
     System.Console.WriteLine("a bread makes man full  ");
    }
}